Source/Diagnostics/ReducedDiags/ColliderRelevant.H

```diff
@@ -0,0 +1,54 @@
+/* Copyright 2023 The WarpX Community
+ *
+ * This file is part of WarpX.
+ *
+ * Authors: Yinjian Zhao, Arianna Formenti
+ * License: BSD-3-Clause-LBNL
+ */
+
+#ifndef WARPX_DIAGNOSTICS_REDUCEDDIAGS_COLLIDERRELEVANT_H_
+#define WARPX_DIAGNOSTICS_REDUCEDDIAGS_COLLIDERRELEVANT_H_
+
+#include "ReducedDiags.H"
+
+#include <map>
+#include <string>
+#include <vector>
+
+/**
+ *  This class mainly contains a function that
+ *  computes the particle extrema of each species.
+ */
+class ColliderRelevant : public ReducedDiags
+{
+public:
+
+    /**
+     * constructor
+     * @param[in] rd_name reduced diags names
+     */
+    ColliderRelevant(std::string rd_name);
+
+    /// name of species
+    std::vector<std::string> m_beam_name;
+
+    /**
+     * This funciton computes the particle extrema
+     *
+     * @param[in] step current time step
+     */
```

Comment:
Do you want to update this docstring with the list of all new reduced diags that this function actually computes? I would also add a `\brief` before the text so that Doxygen can process it correctly.